### PR TITLE
Upgrade tensorflow (binary) to version 2.2.0

### PR DESCRIPTION
**Pull request by jyp**
This PR contains work by @vladmaraev which I submit with his consent.

###### Motivation for this change

Upstream upgrade

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by zowoq** [comment]
@ofborg eval

File: pkgs/development/python-modules/tensorboard-plugin-wit/default.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  };
+
+  # Tests require too many extra dependencies
+  doCheck = false;
+
+  meta = {
+    homepage = https://github.com/tensorflow/addons;
+    description = "Useful extra functionality for TensorFlow 2.x maintained by SIG-addons";
+    license = lib.licenses.bsd3;
+    maintainers = with lib.maintainers; [ vladmaraev ];
```
Comment (jonringer) [inline pkgs/development/python-modules/tensorboard-plugin-wit/default.nix:29]: Please make this into a separate commit:

```
maintainers: add <name>
```

File: pkgs/development/python-modules/tensorboard-plugin-wit/default.nix [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  src = fetchPypi {
+    inherit pname version format;
+    python = "py3";
+    abi = "none";
+    platform = "any";
+    sha256 = "0nv855qm2fav70lndsrv810pqgg41sbmd70fk86wk18ih825yxzf";
+  };
+
+  # Tests require too many extra dependencies
+  doCheck = false;
```
Comment (jonringer) [inline pkgs/development/python-modules/tensorboard-plugin-wit/default.nix:23]: If tests are not available, then please use `pythonImportsCheck` to import the most important modules. This isn't as good as unit tests, but will usually give a good indication of run-time errors. Please see [pythonImportsCheck documentation](https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#using-pythonimportscheck) for more information.

File: pkgs/development/python-modules/tensorflow/2/default.nix [hunk 3]
```diff
@@ -72,7 +72,7 @@ let
 
   tfFeature = x: if x then "1" else "0";
 
-  version = "2.1.0";
+  version = "2.2.0";
```
Comment (matthewbauer) [inline pkgs/development/python-modules/tensorflow/2/default.nix:75]: don't do this - this is for the source-based package

**Comment by mkg20001** [comment]
I've rebased my current PR to upgrade to 2.3.0 against this one. See #98522

Additionally it includes a change to the update script so it no longer requires manually entering the URLs

**Comment by NickCao** [comment]
superseded  by later updates,
